Add explanation of why SCONEPRO is still needed when L4S exists (#82)
mjoras/SCONE-PROTOCL: documents/charter.md

Instead, it would be beneficial, for both the application provider and network operator, to signal network attributes to the application to self-adapt its video traffic to conform to the specified characteristics.
The application provider has the ability to measure end user QoE and therefore can self-adapt with QoE feedback.

L4S is an IETF-specified mechanism where nodes in the network assist endpoints in discovering optimal sending rates through continuous ECN feedback, interpreted by purpose-built congestion control algorithms. 
L4S and ECN operate on RTT timescales, whereas network policies tend to be stable over much longer timescales.
Using L4S to signal this longer-term state is possible, but it conflates separate signals about queuing state and policy.
A signal with longer-term semantics allow senders to temporarily exceed limits, which can be significantly more efficient.

> 

Reviewer: To me this looks fine up until the last sentence, which looks wrong.  As alluded to in the 3rd sentence, L4S signals could be provided by a marking policer that has burst tolerance (thus allowing senders to temporarily exceed limits, i.e. control the rate on longer than RTT timescales), and I don't know what you mean by "significantly more efficient".  Efficient in what sense?
I could be misunderstanding, but it seems to me that the benefits of SCONEPRO are that it: 
1. reduces the need for networks to do traffic identification and per-flow policing for applications that support it (I assume some traffic analysis is still needed to determine whether the application is abiding by the policy limit, but violations can be handled at layer 8 rather than layer 2/3). 
2. allows the application to differentiate between policy-driven rate limitation and congestion-driven rate limitation
3. the combination of 1 & 2 allows an application to more loosely abide by the policy limit without suffering (except at layer 8), while responding to the L2/3 congestion signals with high fidelity to avoid immediate QoE degradation.
You've got a statement about item 2 (though I would have said "congestion state" rather than "queuing state"), but you don't mention items 1 or 3 (unless those are somehow bundled into the words "significantly more efficient").

Author: Thanks for your review. 
Yes, it is possible to temporarily exceed limits with a policer that has some burst tolerance.  With SCONEPRO however, the senders can control burst sizes in terms of application payload boundaries (e.g., video segments), which arguably can have QoE benefits. 

1. Agree
2. Agree
3. Right, the simplest way for the application to 'loosely abide' to the policy would be to simply set a cap on the available quality tracks (assuming ABR video) and not do anything special at the transport layer. I think 3 is what is bundled into to 'significantly more efficient', will try to make the sentence a bit more clear. 

What about: "A signal with longer-term semantics gives senders control of how to exceed limits, which can lead to improved QoE.".

Reviewer: That's better, but I think the distinction that results in improved QoE isn't the long vs. short term semantics aspect, it is the policy vs. congestion aspect. Maybe:

"An unambiguous policy signal gives the application the flexibility to comply with that policy on timescales that work best for the application's QoE, while continuing to respond to congestion signals on RTT timescales."

Author: Thanks, I really like that suggestion!

Instead, it would be beneficial, for both the application provider and network operator, to signal network attributes to the application to self-adapt its video traffic to conform to the specified characteristics.
The application provider has the ability to measure end user QoE and therefore can self-adapt with QoE feedback.

L4S is a recently standardized mechanism where nodes in the network assist endpoints in discovering optimal sending rates through continuous ECN feedback, interpreted by purpose-built congestion control algorithms. 

> However, L4S alone is not sufficient to communicate network properties related to video self-adaptation. 

Reviewer 2: ```suggestion
```

> L4S and ECN operate on RTT timescales, reflecting the current state of a network buffer, whereas the network properties related to video traffic reflect policies that tend to be stable over orders of magnitude longer timescales. 

Reviewer 3: ```suggestion
L4S and ECN operate on RTT timescales, reflecting the current state of a network buffer, whereas the network properties that allow video traffic adaptation reflect policies that tend to be stable over orders of magnitude longer timescales. 
```

Reviewer 2: I think I prefer what Marcus wrote than this suggestion, Spencer.  Still, it could be more brief.

Also

```suggestion
L4S and ECN operate on RTT timescales, whereas network policies tend to be stable over much longer timescales.
```

Author: I like Martins proposal. It's brief and straight to the point.

> It is possible to construct an L4S-capable traffic policer that marks packets that do not conform to some policy instead of dropping or queueing them. 
> Such a policer is not explicitly distinguishable from a network node that experiences congestion. 

Reviewer 3: ```suggestion
While it is possible to construct an L4S-capable traffic policer that marks packets that do not conform to some policy instead of dropping or queueing them, such a policer does not empower a video receiver to adapt the resources it requests. 
The receiver is limited to transport-level adaptation, rather than application-level adaptation, and cannot do more than it does when a network node along the path between sender and receiver experiences congestion. 
```

Reviewer 3: It may be worth pointing out that ECN and L4S use IP header fields that may be rewritten at network boundaries, but I'll let you Do The Right Thing.

Author: That is a good point, but I'm worried that we'll inflate the charter text if we start listing all kinds of potential issues.

> It is possible to construct an L4S-capable traffic policer that marks packets that do not conform to some policy instead of dropping or queueing them. 
> Such a policer is not explicitly distinguishable from a network node that experiences congestion. 

Reviewer 2: I would maybe just say:

```suggestion
Using L4S to signal this longer-term state is possible, but it conflates separate signals about queuing state and policy.
```

> Furthermore, a more long-term signal that is consumed by the application layer allows for efficient bursting of video segments, with burst sizes controlled by the sending entity rather than on-path network policers. 

Reviewer 2: ```suggestion
A signal with longer-term semantics allow senders to temporarily exceed limits, which can be significantly more efficient.
```

Author: Thanks, this is a great way of phrasing it.
